strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseController.java

```diff
@@ -174,4 +188,120 @@
         }
     }
     
+    @ApiOperation(value = "Used to browse the configured storages")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = ""),
+                            @ApiResponse(code = 400, message = "An error occurred.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(method = RequestMethod.GET, 
+                    produces = MediaType.TEXT_HTML_VALUE)
+    public void browseStorages(HttpServletRequest request,
+                               HttpServletResponse response)
+    {
+        logger.debug("Requested browsing for storages");
+
+        getDirectoryListing(request, response);
```

Comment:
This should be done as MVC pattern.
you should use same logic here and in `browse()` method to get storages model. Within this method you should just produce HTML content based on such model.